### DISCUSSION TITLE
Reconsider supporting #noqa now that Pyright does syntax-checking (Issue #144)

### DISCUSSION THREAD

**Discussion by J-Sorenson** [microsoft/pyright#7607] [Ideas]
We updated our VS Code/PyLance/Pyright installations recently, and we're now getting flooded with warnings from vendor libraries because pyright insists on checking unbound variables, missing members, etc on lines marked with "# noqa."  This was originally brought up in issue #144 , but was dismissed and closed by @erictraut because "Pyright doesn't perform any linting (code style) checks, only type checks."  That's apparently no longer the case, and now Pyright is checking syntax were it should not.  "noqa" is a well known standard for linters, and using "# type: ignore" would be incorrect (since it has nothing to do with type).   Either we re-open the original issue, or I propose creating a new issue that references the original.  Any comments beforehand?

**Comment by erictraut** [comment]
Pyright is a standards-compliant type checker, and `#noqa` is not part of the typing spec. If you want to suppress diagnostics generated by a type checker on a given line, you can use a [# type:  ignore](https://typing.readthedocs.io/en/latest/spec/directives.html#type-ignore-comments) comment, whose behavior is specified in the typing spec.

Pyright also offers extensive configuration capabilities and fine-grained control over diagnostics. You can disable specific diagnostic rules on a per-project or per-file basis.

As I've stated previously, pyright is not a linter. It doesn't enforce code style rules. If you want linter functionality, you should use tools like pylint or ruff, which work in conjunction with type checkers.

We have no plans to support `#noqa`. Any feature request along these lines will be immediately rejected, so please don't bother opening a new request or re-opening the original.

**Reply by J-Sorenson** [reply]
Hey, Eric.  I think many would argue that reportOverlappingOverload, reportMissingSuperCall, and reportPrivateUsage are not type checks.  Pyright scope has expanded in the last year.  In any case, I'm not going to add Pyright-specific lines to the top of all the vendor files.  I want Pyright to scan those files so that we can get the types correct when using those libraries, but not if it's going to add a lot of noise on things unrelated to type-checking.  
Based on your very firm stance on the #noqa issue (type-checking is not quality assurance?), it sounds like my best option is to update the pyrightconfig.json to disable all non-type-related checks from Pyright and install/use Pyflakes for those instead.  Fair enough.
